### PR TITLE
[WIP][NEEDS REVIEW][DNM] ITV Sleeping Slug Trader Ship - Off-Station Ship

### PR DESCRIPTION
**Pull request by Enzo-Leon**
This is a rather massive update that includes a lot to the existing code, as well as working on attempting to modularize this 100% if possible. This is BY NO MEANS READY FOR EVEN TEST MERGING YET AND WILL PROBABLY STILL HAVE SOME TIME TO GO ON IT. THIS ENTIRE THING WAS MADE IN JUST A FEW DAYS. THERE ARE STILL THINGS NEEDING ATTENTION AND STUFF THAT'S BEEN OVERLOOKED/MISSED/DUPLICATED ON MISTAKE.

((Okay apparently it passed checks already so IN THEORY IT WOULD RUN JUST FINE BUT IT WOULD BE MORE FOR TESTING THEN ANYTHING FOR THE LIVE SERVER))

To-do's Still to... do.
---
- Finer details
- Loot spawn tweaking (after feedback of initial review, and immediate questions/concerns/comments)
- Decals overview and tweaking for best appearance
- Additional equipment added and adjusted (Suit cyclers, weapons adjustments, etc.)
- POSSIBLE expansion to the current layout to add maintenance (Since this isn't a primary station job, maintenance didn't seem as needed for this.)

The ship
---
wip_ship

The Jobs
---
Sleeping Slug Captain, Guard, Doctor, Pilot, and Engineer. Currently, each current job is restricted to 1 slot each for a total of 5 crew for the off-station map. The ship is able to entirely run with just one person at worst and provides enough resources to where they won't run into an issue in their round.

The Goal
---
A focus heavily on trade and RP interactions. While exploration would be possible, the ship will start with more than enough ample gear and supplies to conduct trade. This is an active and known concern about introducing something like this to the server. While it will bring a lot to the environment and help start paving the way for more overmaps based content (There's a LOT of stuff that we can do with this) I feel that concern will be mild at best, I hope. 

For the time being with only SIF as an exploration zone that even non-exploration crew are free to roam there is not the ship will be able to do aside from RP matters (and this will also give events staff a new playground to mess with for events, or stuff to cause a distress beacon to get utilized). 

A future PR unrelated to this will be opened over the coming weeks for more exploration content (and will try to ensure that the maximum amount of players can interact with it. Similar to the UnderTether that Citadel-RP used to have a while back as a test project.)

### CODE REVIEW

**Comment by Rykka-Stormheart** [comment]
Hyped asf for this. Well done on what you have so far! <3

**Comment by cadyn** [comment]
Hey! Just a little side note, if it's WIP you should probably mark the PR as draft. <3

File: maps/southern_cross/submaps/_southern_cross_submaps.dm [hunk 1]
```diff
@@ -29,6 +29,20 @@
 	new /datum/random_map/noise/ore(null, 1, 1, Z_LEVEL_BELT, 64, 64)         // Create the mining ore distribution map.
 
 */
+
+/datum/map_template/sc_lateload/offmap/sleeping_slug
+	name = "Offmap Ship - Sleeing Slug"
```
Comment (cadyn) [inline maps/southern_cross/submaps/_southern_cross_submaps.dm:34]: Sleeing slug lol. Typo.

File: code/modules/clothing/under/miscellaneous_ch.dm [hunk 2]
```diff
@@ -8,3 +8,55 @@
 
 /obj/item/clothing/under/sexybunny_white/sexybunny_black
 	icon_state = "sexybunny_black"
+
+//////////////////////TALON JUMPSUITS//////////////////////
```
Comment (cadyn) [inline code/modules/clothing/under/miscellaneous_ch.dm:12]: Talon should be replaced by sleeping slug here I'm guessing lol

File: code/game/machinery/suit_cycler_datums_ch.dm [hunk 3]
```diff
@@ -0,0 +1,44 @@
+/datum/suit_cycler_choice/department/sleeping_slug/crew
+	name = "Talon Crew"
```
Comment (cadyn) [inline code/game/machinery/suit_cycler_datums_ch.dm:2]: More Talon references spotted. All throughout this file.

File: code/modules/clothing/spacesuits/void/event_ch.dm [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+/obj/item/clothing/head/helmet/space/void/refurb/medical/alt/sleeping_slug
+	name = "sleeping slug medical voidsuit bubble helmet"
+	camera_networks = list(NETWORK_SLEEPING_SLUG_HELMETS)
+
+/obj/item/clothing/suit/space/void/refurb/medical/sleeping_slug
+	name = "sleeping slug medical voidsuit"
+
+/obj/item/clothing/head/helmet/space/void/refurb/marine/sleeping_slug
+	name = "sleeping slug marine's voidsuit helmet"
+	desc = "A refurbished early contact era voidsuit helmet of human design. These things aren't especially good against modern weapons but they're sturdy, incredibly easy to come by, and there are lots of spare parts for repairs. The visor has a bad habit of fogging up and collecting condensation, but it beats sucking hard vacuum. The blue markings indicate this as the marine/guard variant. \"ITV TALON\" has been stamped onto the sides of the helmet."
```
Comment (cadyn) [inline code/modules/clothing/spacesuits/void/event_ch.dm:32]: Mentions ITV Talon on a few of these, in the description.

**Review by cadyn — COMMENTED** [review]
All the code looks good to me. I'm guessing the large portion of this was copied from the talon which is 100% the right move here, but I spotted a bunch of lingering references to the talon, and have pointed them out in some comments.

I think the big part of this, though, is not so much the quality of code but how this will impact gameplay. A lot of people who've come from Virgo know that Talon could be both a blessing and a curse, and lots of drama tended to spawn from it. 

Personally, I'm neutral. I can see it going both ways, but I think if we do add an offmap spawn, we need to be very careful about our policy and lore for it. I think a big part of what makes Talon questionable on virgo is the fact they're answerable to no-one. If we're going to add an offmap spawn, we should probably implement a sort of CentComm equivalent for them.

Also to ensure that the ITV Sleeping slug doesn't end up interfering with the crew in a negative way, we should probably have them be lorewise have some sort of trade agreement or alliance with NT and the southern cross.

Another good idea is probably to make the role whitelisted. Same goes with any new role, especially those which have the potential to be abused.

**Comment by Enzo-Leon** [comment]
> All the code looks good to me. I'm guessing the large portion of this was copied from the talon which is 100% the right move here, but I spotted a bunch of lingering references to the talon, and have pointed them out in some comments.

Yeah I noticed a few of those getting missed when I was replacing and updating stuff. At least it seemed to be mainly comments stuff and I can do a pass through and fix those.

> I think the big part of this, though, is not so much the quality of code but how this will impact gameplay. A lot of people who've come from Virgo know that Talon could be both a blessing and a curse, and lots of drama tended to spawn from it.
>
> Personally, I'm neutral. I can see it going both ways, but I think if we do add an offmap spawn, we need to be very careful about our policy and lore for it. I think a big part of what makes Talon questionable on virgo is the fact they're answerable to no-one. If we're going to add an offmap spawn, we should probably implement a sort of CentComm equivalent for them.
> 
> Also to ensure that the ITV Sleeping slug doesn't end up interfering with the crew in a negative way, we should probably have them be lorewise have some sort of trade agreement or alliance with NT and the southern cross.

God this is 100% my same thought and concern on this. I wanna give us an option for space stuff, and maybe a way for admins to add some lost in space, damaged/etc style mini-events since we have working ships in the exploration "carrier".
 
> Another good idea is probably to make the role whitelisted. Same goes with any new role, especially those which have the potential to be abused.

Honestly, I would prefer this too, but not sure what staff wanna do on that. I think ensuring this is behind a whitelist would be good idea since then we can enforce a rule like "To be part of the whitelist you agree to not bum rush exploration content, and more so since we have so little of it currently."

**Comment by Razgriz1032** [comment]
Any progress here, or can I close this?

**Comment by Darlantanis** [comment]
Assuming this is abandoned given it's been untouched for over a year and a half.

**Pull request closed without merging**